plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -654,10 +669,32 @@ func (p *Plugin) validateNodeServiceAccountAudience(ctx context.Context, tr *aut
 	if err != nil {
 		return fmt.Errorf("error validating audience %q: %w", requestedAudience, err)
 	}
-	if !foundAudiencesInPodSpec {
-		return fmt.Errorf("audience %q not found in pod spec volume", requestedAudience)
+	if foundAudiencesInPodSpec {
+		return nil
 	}
-	return nil
+
+	attrs := authorizer.AttributesRecord{
+		User:            userInfo, // this is the user info of the node requesting the token
+		Verb:            "request-serviceaccounts-token-audience",
+		Namespace:       tr.Namespace,
```

Comment:
let's use `a admission.Attributes` here to get namespace and name, since those are definitely derived from the URL and don't depend on the content of the TokenRequest being submitted